src/ui_components/settings.js

```diff
@@ -102,6 +102,13 @@ class Settings extends React.Component {
         this.createLog('Map Zooming', e.target.checked);
     }
 
+    //handle toggle of hiding/showing location
+    handleLocationPublicChange(e) {
+        //this.props.locationPublic(e.target.checked);
```

Comment:
This seems almost right! I think you're just using the wrong prop name on this line (you want onLocationPublic instead of just locationPublic). Do you want to look at this together?